frontend/public/components/resource-quota.jsx

```diff
@@ -30,11 +124,77 @@ const Row = ({obj: rq}) => <div className="row co-resource-list__item">
   </div>
 </div>;
 
+const CapacityIcon = ({capacity}) => {
+  let capacityIconClass = 'pficon-unknown';
+  if (capacity === 0) {
+    capacityIconClass = 'fa-circle-thin co-m-resource-quota-empty';
+  } else if (capacity > 0 && capacity < .5) {
+    capacityIconClass = 'pficon-resources-almost-empty';
+  } else if (capacity >= .5 && capacity < 1){
+    capacityIconClass = 'pficon-resources-almost-full';
+  } else if (capacity === 1) {
+    capacityIconClass = 'pficon-resources-full';
+  } else if (capacity > 1) {
+    capacityIconClass = 'pficon-warning-triangle-o';
+  }
+  const classPrefix = capacityIconClass.startsWith('pficon') ? 'pficon' : 'fa';
+  return <i className={`${classPrefix} ${capacityIconClass}`} aria-hidden="true" />;
+};
+
+const getCapacity = (used, max, resourceType) => {
+  let capacity = 0;
+  if (max && used) {
+    let unitType = units.getUnitType(max);
+    if (!unitType) {
+      unitType = units.getUnitType(used);
+    }
+    // max cpu values do not have units, convert number and add unit for capacity calculation
+    // Ex. converts '1' to '1000m'
+    let maxStr = max;
+    let maxNum = _.toNumber(max);
+    if (_.includes(resourceType, '.cpu') && unitType && !_.isNaN(maxNum)) { // skip if max cpu already has a unit
+      maxStr = `${maxNum * 1000}m`;
+    }
+    maxNum = units.dehumanize(maxStr, unitType).value;
+    const usedNum = units.dehumanize(used, unitType).value;
+    capacity = usedNum / maxNum;
+  }
+  return capacity;
+};
+
+const ResourceUsageRow = ({quota, resourceType}) => {
+  const max = quota.status.hard ? quota.status.hard[resourceType] : quota.spec.hard[resourceType];
```

Comment:
```suggestion
  const max = _.get(quota, ['status', 'hard', resourceType]) || _.get(quota, ['spec', 'hard', resourceType]);
```

(And similar for below.)

Reply:
fixed